src/olympia/lib/git.py

```diff
@@ -397,7 +406,11 @@ def get_author(self, user=None):
 
     def find_or_create_branch(self, name):
         """Lookup or create the branch named `name`"""
-        branch = self.git_repository.branches.get(name)
+        try:
+            branch = self.git_repository.branches.get(name)
+        except pygit2.GitError:
+            raise BrokenRefError('Reference for branch "{}" is '
```

Comment:
Probably should log here as well

Reply:
Yep. This error is logged in the caller code but we might also want to log it here as well.